### PR TITLE
refactor: promisify create tab

### PR DESCRIPTION
**Pull request by smoralesd**
#### Description of changes

This is part of a long-run refactor to promisify all (possible) `chrome.*` calls.

In this case, promisifying `createTab`.

#### Pull request checklist

- [ ] Addresses an existing issue: #0000
- [x] Ran `yarn fastpass`
- [x] Added/updated relevant unit test(s) (and ran `yarn test`)
- [x] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] PR title *AND* final merge commit title both start with a semantic tag (`fix:`, `chore:`, `feat(feature-name):`, `refactor:`). Check workflow guide at: `<rootDir>/docs/workflow.md`
- [ ] (UI changes only) Added screenshots/GIFs to description above
- [ ] (UI changes only) Verified usability with NVDA/JAWS

### CODE REVIEW

File: src/background/actions/shortcuts-page-action-creator.ts [hunk 1]
```diff
@@ -18,8 +19,8 @@ export class ShortcutsPageActionCreator {
         this.interpreter.registerTypeToPayloadCallback(Messages.Shortcuts.ConfigureShortcuts, this.onConfigureShortcuts);
     }
 
-    private onConfigureShortcuts = (payload: BaseActionPayload): void => {
-        this.shortcutsPageController.openShortcutsTab();
+    private onConfigureShortcuts = async (payload: BaseActionPayload): Promise<void> => {
+        await this.shortcutsPageController.openShortcutsTab();
```
Comment (dbjorge) [inline src/background/actions/shortcuts-page-action-creator.ts]: This has the side effect of preventing the telemetry from being sent in the event that opening the shortcuts tab encounters an error. Is that desirable?

File: src/tests/unit/tests/background/shortcuts-page-controller.test.ts [hunk 2]
```diff
@@ -1,17 +1,35 @@
...
+
+        testSubject = new ShortcutsPageController(browserAdapterMock.object);
+    });
+
+    it('opens the shortcuts tab', async () => {
+        browserAdapterMock
+            .setup(adapter => adapter.createActiveTab(ShortcutsPageController.configureCommandTabUrl))
+            .returns(() => Promise.resolve({} as Tabs.Tab));
+
+        await expect(testSubject.openShortcutsTab()).resolves.toBe(undefined);
```
Comment (dbjorge) [inline src/tests/unit/tests/background/shortcuts-page-controller.test.ts]: Should this be verifying that the call to `createActiveTab` actually happens?

File: src/tests/unit/tests/issue-filing/common/create-file-issue-handler.test.ts [hunk 3]
```diff
@@ -1,45 +1,60 @@
...
+        browserAdapterMock = Mock.ofType<BrowserAdapter>(undefined, MockBehavior.Strict);
+    });
+
+    it('properly files an issue', async () => {
+        browserAdapterMock.setup(adapter => adapter.createActiveTab(urlStub)).returns(() => Promise.resolve({} as Tabs.Tab));
 
         const testSubject = createFileIssueHandler(urlProviderMock.object, settingsGetterMock.object);
 
-        testSubject(browserAdapterMock.object, serviceMap, issueData, environmentInfoStub);
+        await expect(testSubject(browserAdapterMock.object, serviceMap, issueData, environmentInfoStub)).resolves.toBe(undefined);
```
Comment (dbjorge) [inline src/tests/unit/tests/issue-filing/common/create-file-issue-handler.test.ts]: Should this be verifying that the correct side effects happen?

File: src/tests/unit/tests/popup/components/file-url-unsupported-message-panel.test.tsx [hunk 4]
```diff
@@ -28,11 +29,11 @@ describe('FileUrlUnsupportedMessagePanel', () => {
         expect(wrapper.getElement()).toMatchSnapshot();
     });
 
-    it('has a NewTabLink that uses createTab to open the manage extension page', () => {
+    it('has a NewTabLink that uses createTab to open the manage extension page', async () => {
```
Comment (dbjorge) [inline src/tests/unit/tests/popup/components/file-url-unsupported-message-panel.test.tsx]: "uses create**Active**Tab"?

Comment (smoralesd) [reply src/tests/unit/tests/popup/components/file-url-unsupported-message-panel.test.tsx]: updated

File: src/tests/unit/tests/issue-filing/common/create-file-issue-handler.test.ts (hunk 3 above)
Comment (smoralesd) [reply src/tests/unit/tests/issue-filing/common/create-file-issue-handler.test.ts]: good catch.

File: src/tests/unit/tests/background/shortcuts-page-controller.test.ts (hunk 2 above)
Comment (smoralesd) [reply src/tests/unit/tests/background/shortcuts-page-controller.test.ts]: updated

File: src/background/actions/shortcuts-page-action-creator.ts (hunk 1 above)
Comment (smoralesd) [reply src/background/actions/shortcuts-page-action-creator.ts]: Good catch!

Comment (smoralesd) [reply src/background/actions/shortcuts-page-action-creator.ts]: Updated.

**Pull request merged**